### PR TITLE
Add Get-GcLogSink cmdlet

### PR DESCRIPTION
**Pull request by quoctruong**
This change is [Reviewable](https://reviewable.io/reviews/googlecloudplatform/google-cloud-powershell/420)

### CODE REVIEW

File: Google.PowerShell.IntegrationTests/Logging/Logging.GcLogSink.Tests.ps1 [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    
+
+    AfterAll {
+        gcloud beta logging sinks delete $sinkName --quiet 2>$null
+        gcloud beta logging sinks delete $secondSinkName --quiet 2>$null
+    }
+
+    It "should work without any parameters" {
+        $sinks = Get-GcLogSink
+        $sinks.Count - $previousCount | Should Be 2
```
Comment (iantalarico) [inline Google.PowerShell.IntegrationTests/Logging/Logging.GcLogSink.Tests.ps1]: Is this prone to failure? Is this thread safe?

File: Google.PowerShell.IntegrationTests/Logging/Logging.GcLogSink.Tests.ps1 [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        $secondSink.Destination | Should BeExactly $destinationTwo
+        $secondSink.OutputVersionFormat | Should BeExactly V2
+        $secondSink.Filter | Should BeNullOrEmpty
+        $secondSink.WriterIdentity | Should Not BeNullOrEmpty
+    }
+
+    It "should work with -Sink parameter" {
+        $firstSink = Get-GcLogSink -Sink $sinkName
+        $firstSink.Name | Should BeExactly "$sinkName"
+        $firstSink | Should Not BeNullOrEmpty
```
Comment (iantalarico) [inline Google.PowerShell.IntegrationTests/Logging/Logging.GcLogSink.Tests.ps1]: Should this be before the $firstSink.Name check?

File: Google.PowerShell.IntegrationTests/Logging/Logging.GcLogSink.Tests.ps1 [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        $firstSink.Destination | Should BeExactly $destination
+        $firstSink.OutputVersionFormat | Should BeExactly V1
+        $firstSink.Filter | Should BeExactly $logFilter
+        $firstSink.WriterIdentity | Should Not BeNullOrEmpty
+
+        # Should work with an array of sinks names.
+        $sinks = Get-GcLogSink -Sink $sinkName, $secondSinkName
```
Comment (iantalarico) [inline Google.PowerShell.IntegrationTests/Logging/Logging.GcLogSink.Tests.ps1]: Maybe a separate test for this?

File: Google.PowerShell/Logging/GcLogCmdlets.cs [hunk 4]
```diff
@@ -90,6 +90,18 @@ protected string PrefixProject(string logName, string project)
         }
 
         /// <summary>
+        /// Prefix projects/{project id}/sinks to sinkName if not present.
+        /// </summary>
+        protected string PrefixProjectToSinkName(string sinkName, string project)
+        {
+            if (!string.IsNullOrWhiteSpace(sinkName) && !sinkName.StartsWith($"projects/{project}/sinks"))
+            {
+                sinkName = $"projects/{project}/sinks/{sinkName}";
```
Comment (iantalarico) [inline Google.PowerShell/Logging/GcLogCmdlets.cs]: Aren't their resources that will do this formatting for you?  I think for Logs their is LogName?

File: Google.PowerShell/Logging/GcLogSinkCmdlets.cs [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+using Google.Apis.Logging.v2.Data;
+using Google.PowerShell.Common;
+using System.Management.Automation;
+using System.Net;
+
+namespace Google.PowerShell.Logging
+{
+    /// <summary>
+    /// <para type="synopsis">
+    /// Retrieves StackDriver Log Sinks.
```
Comment (iantalarico) [inline Google.PowerShell/Logging/GcLogSinkCmdlets.cs]: nit: Stackdriver

and throughout the PR

File: Google.PowerShell/Logging/GcLogSinkCmdlets.cs [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+    /// <para type="link" uri="(https://cloud.google.com/logging/docs/api/tasks/exporting-logs#about_sinks)">
+    /// [Log Sinks]
+    /// </para>
+    /// </summary>
+    [Cmdlet(VerbsCommon.Get, "GcLogSink")]
+    public class GetGcLogSinkCmdlet : GcLogCmdlet
+    {
+        /// <summary>
+        /// <para type="description">
+        /// The project to check for sinks. If not set via PowerShell parameter processing, will
```
Comment (iantalarico) [inline Google.PowerShell/Logging/GcLogSinkCmdlets.cs]: to check for sinks in?

File: Google.PowerShell/Logging/GcLogCmdlets.cs (hunk 4 above)
Comment (quoctruong) [inline Google.PowerShell/Logging/GcLogCmdlets.cs]: Hi @iantalarico, what do you mean by resources that will do this formatting? Like a .NET library?

Comment (iantalarico) [inline Google.PowerShell/Logging/GcLogCmdlets.cs]: Yea but I just saw you are using the apis directly not the veneers.  Is there a reason for that?

Comment (quoctruong) [inline Google.PowerShell/Logging/GcLogCmdlets.cs]: I think when we started the PowerShell module, the veneers are not available yet. Is there a big performance difference between the two?

**Comment by quoctruong** [comment]
Thanks!

File: Google.PowerShell/Logging/GcLogCmdlets.cs (hunk 4 above)
Comment (iantalarico) [inline Google.PowerShell/Logging/GcLogCmdlets.cs]: I don't think their is any performance difference.  It just makes using the APIs easier.

**Pull request merged**